colossalai/zero/sharded_optim/sharded_optim_v2.py

```diff
@@ -193,15 +197,20 @@ def step(self, *args, **kwargs):
 
         self._point_param_fp16_to_master_param()
 
-        self._logger.debug(
-            f"Before step ShardedOptimizerV2 consumes {self.get_memory_usage()[0] / 1e6} MB CUDA Memory, {self.get_memory_usage()[1] / 1e6} MB CUDA Memory!",
-            ranks=[0])
+        if self._verbose:
+            gpu_mem, cpu_mem = self.get_memory_usage()
+            self._logger.debug(
+                f"Before step ShardedOptimizerV2 consumes {gpu_mem / 1e6} MB CUDA Memory, {cpu_mem / 1e6} MB CUDA Memory!",
```

Comment:
You imply that we shall not call function to build the str for logging?

Reply:
Yeah, since the functions are always called whether the string would be printed.